pandas/core/indexes/datetimes.py

```diff
@@ -649,6 +653,20 @@ def _simple_new(cls, values, name=None, freq=None, tz=None,
         result._reset_identity()
         return result
 
+    def _assert_tzawareness_compat(self, other):
+        # adapted from _Timestamp._assert_tzawareness_compat
+        other_tz = getattr(other, 'tzinfo', None)
```

Comment:
this is odd to use ``tzinfo``, these are already wrapped scalars, or an index type, so ``.tz`` is appropriate

Reply:
That was my first thought too but `other` could be a raw `datetime`.

Reply:
use .tz, you can simply wrap ``other = pd.Timestamp(other)``